pkg/lib/authenticationflow/milestone.go

```diff
@@ -17,9 +17,13 @@ func FindMilestone[T Milestone](w *Flow) (T, bool) {
 	return findMilestone[T](w, false)
 }
 
-// This function only find milestones in the provided flow,
-// and will not nest into subflows
-func FindMilestoneInCurrentFlow[T Milestone](w *Flow) (T, bool) {
+// FindMilestoneInCurrentFlow find the last milestone in the flow.
+// It does not recur into sub flows.
+// If the found milestone is a node, then the returned flows is the same as flows.
+// If the found milestone is a intent, then the returned flows is Nearest=intent.
+func FindMilestoneInCurrentFlow[T Milestone](flows Flows) (T, Flows, bool) {
```

Comment:
In this commit, it seems the new returned `Flows` is always ignored. Did you find any places that we need this new `Flows`?

Reply:
Not always, it is need sometimes, like in https://github.com/authgear/authgear-server/pull/4372/commits/6d78a3dccf400c23c8e1b52f614d82561d732df0#diff-ea1ed68baf6a6ca64bbdb127d2f22167bb59c0a26052e499c514f7e0cf06645dR59

The returned flows will have Nearest updated to match the root of the returned milestone. We can then use the returned milestone and the returned flows to call FindMilestoneInCurrentFlow. It is a more structured find. The previous FindMilestone finds too deep milestone that is irrelevant to the finder.